app/main.py

```diff
@@ -1,3 +1,38 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        data = json.load(file)
+    fuel_price = data["FUEL_PRICE"]
+    customers = [
+        Customer(
+            customer["name"],
+            customer["product_cart"],
+            customer["location"],
+            customer["money"],
+            customer["car"])
+        for customer in data["customers"]]
+    shops = [
+        Shop(
+            shop["name"],
+            shop["location"],
+            shop["products"]
+        )
+        for shop in data["shops"]
+    ]
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        trips_price = sorted(
+            [shop.count_trip_price(customer, fuel_price) for shop in shops],
+            key=lambda x: x.trip_price
+        )
+        cheapest_shop = trips_price[0]
```

Comment:
You can use `min()` to determine the cheapest shop. I think it will be more readable for this case